core/src/main/java/com/linecorp/armeria/server/ServerConfig.java

```diff
@@ -104,6 +103,8 @@
 
         this.serviceLoggerPrefix = ServiceConfig.validateLoggerName(serviceLoggerPrefix, "serviceLoggerPrefix");
 
+        maxConnectionHandler = new MaxConnectionHandler(maxConnections);
```

Comment:
Let's create a handler later when a pipeline is configured. This class is just meant to provide the configuration parameters, so I think it's enough to have `int maxConnections()` like before. Also, it's not a good idea to expose a public method that returns a handler, which allows a user to call arbitrary methods on it.